### PR TITLE
GR-1419 dashi median to mean insert size

### PR DESCRIPTION
**Pull request by mengjenn**
Jira ticket or GitHub issue: [[GR-1419] Switch Dashi Median Insert Size graphs to Mean Insert Size](https://jira.oicr.on.ca/browse/GR-1419)

- [x] Updates CHANGELOG.md
- [ ] Updates dev docs if applicable

### CODE REVIEW

File: application/dash_application/views/call_ready_tar.py [hunk 1]
```diff
@@ -293,21 +293,18 @@ def generate_callability(df, graph_params):
...
-def generate_median_insert_size(df, graph_params):
+def generate_mean_insert_size(df, graph_params):
     return CallReadySubplot(
-        "Median Insert Size with 10/90 Percentile",
+        "Mean Insert Size + Intron",
         df,
-        lambda d: d[BAMQC_COL.InsertMedian],
-        "Base Pairs",
+        lambda d: d[BAMQC_COL.InsertMean],
+        "Mean Insert Size + Intron",
```
Comment (avarsava) [inline application/dash_application/views/call_ready_tar.py]: This appears to be the only report where the 'Base Pairs' y-axis label changed, is this correct?

File: application/dash_application/views/single_lane_tar.py [hunk 2]
```diff
@@ -265,16 +265,16 @@ def generate_on_target_reads(current_data, graph_params):
     )
 
 
-def generate_median_insert_size(current_data, graph_params):
+def generate_mean_insert_size(current_data, graph_params):
     return SingleLaneSubplot(
-        "Median Insert Size with 10/90 Percentile",
+        "Mean Insert Size with 10/90 Percentile",
```
Comment (avarsava) [inline application/dash_application/views/single_lane_tar.py]: This report seems to have a mix of the '+ Intron' and percentile labeling?

File: application/dash_application/utility/plot_builder.py [hunk 3]
```diff
@@ -101,8 +101,8 @@
     BEDTOOLS_COL.Coverage10Percentile: '10 Percentile Coverage: ',
     BAMQC_COL.CoverageMedian90Percentile: '90 Percentile Median Coverage: ',
     BAMQC_COL.CoverageMedian10Percentile: '10 Percentile Median Coverage: ',
-    BAMQC_COL.Insert90Percentile: '90 Percentile Median Insert: ',
-    BAMQC_COL.Insert10Percentile: '10 Percentile Median Insert: ',
+    BAMQC_COL.Insert90Percentile: '90 Percentile Mean Insert: ',
+    BAMQC_COL.Insert10Percentile: '10 Percentile Mean Insert: ',
```
Comment (slazicoicr) [inline application/dash_application/utility/plot_builder.py]: The percentile bars should be removed. QM-024 does not mention distribution information (percentile, standard deviation). I didn't write this in the original ticket, so oversight on my part.

File: application/dash_application/views/call_ready_tar.py [hunk 4]
```diff
@@ -192,8 +192,8 @@ def get_merged_ts_data():
 # TODO: Look at ALL the reports for common labels
 cutoff_callability_label = "Callability minimum"
 initial["cutoff_callability"] = 50
-cutoff_insert_median_label = sidebar_utils.insert_median_cutoff_label
-initial["cutoff_insert_median"] = 150
+cutoff_insert_mean_label = "Insert Size Mean + Intron"
```
Comment (slazicoicr) [inline application/dash_application/views/call_ready_tar.py]: Introns affect the insert size of RNA (single lane and call ready) only. Targeted sequencing and whole genome views are "Insert Size Mean".

File: application/dash_application/views/call_ready_wgs.py [hunk 5]
```diff
@@ -170,8 +170,8 @@ def get_merged_wgs_data():
 # TODO: This is supposed to depend on Coverage being 80x/30x?
 cutoff_callability_label = "Callability minimum"
 initial["cutoff_callability"] = 50
-cutoff_insert_median_label = sidebar_utils.insert_median_cutoff_label
-initial["cutoff_insert_median"] = 150
+cutoff_insert_mean_label = "Insert Size Mean + Intron"
```
Comment (slazicoicr) [inline application/dash_application/views/call_ready_wgs.py]: Introns are not a concern with whole genome sequencing.

**Review by slazicoicr — CHANGES_REQUESTED** [review]
I get a exception when I open the Single-Lane cfMeDIP view
```
line 254, in generate_mean_insert_size
    bar_positive=INSERT_COL.InsertMean90Percentile,
AttributeError: type object 'InsertSizeMetricsColumn' has no attribute 'InsertMean90Percentile'
```

File: application/dash_application/views/call_ready_tar.py [hunk 6]
```diff
@@ -192,8 +192,8 @@ def get_merged_ts_data():
 # TODO: Look at ALL the reports for common labels
 cutoff_callability_label = "Callability minimum"
 initial["cutoff_callability"] = 50
-cutoff_insert_median_label = sidebar_utils.insert_median_cutoff_label
-initial["cutoff_insert_median"] = 150
+cutoff_insert_mean_label = sidebar_utils.insert_mean_cutoff_label
+initial["cutoff_insert_mean"] = 100
```
Comment (slazicoicr) [inline application/dash_application/views/call_ready_tar.py]: Cutoff stays at 150.

**Comment by avarsava** [comment]
> I get a exception when I open the Single-Lane cfMeDIP view
> 
> ```
> line 254, in generate_mean_insert_size
>     bar_positive=INSERT_COL.InsertMean90Percentile,
> AttributeError: type object 'InsertSizeMetricsColumn' has no attribute 'InsertMean90Percentile'
> ```

also getting this, even with the latest changes

File: application/dash_application/views/single_lane_cfmedip.py [hunk 7]
```diff
@@ -242,17 +242,17 @@ def generate_percent_pf_reads_aligned(current_data, graph_params):
...
-        lambda d: d[INSERT_COL.MedianInsertSize],
+        lambda d: d[INSERT_COL.MeanInsertSize],
         "Base Pairs",
         graph_params["colour_by"],
         graph_params["shape_by"],
         graph_params["shownames_val"],
-        bar_positive=INSERT_COL.InsertMedian90Percentile,
-        bar_negative=INSERT_COL.InsertMedian10Percentile,
+        bar_positive=INSERT_COL.InsertMean90Percentile,
+        bar_negative=INSERT_COL.InsertMean10Percentile,
```
Comment (slazicoicr) [inline application/dash_application/views/single_lane_cfmedip.py]: For consistency, these bars can go as well.

**Comment by mengjenn** [comment]
> > I get a exception when I open the Single-Lane cfMeDIP view
> > ```
> > line 254, in generate_mean_insert_size
> >     bar_positive=INSERT_COL.InsertMean90Percentile,
> > AttributeError: type object 'InsertSizeMetricsColumn' has no attribute 'InsertMean90Percentile'
> > ```
> 
> also getting this, even with the latest changes

Gah, it's the line above calling InsertMean90Percentile (which doesn't exist anymore). Removing it _should_ fix the issue.

**Review by slazicoicr — CHANGES_REQUESTED** [review]

**Review by avarsava — APPROVED** [review]

**Review by slazicoicr — APPROVED** [review]

**Pull request merged**